tsl/test/sql/data_node.sql

```diff
@@ -585,8 +585,17 @@ SELECT * FROM timescaledb_experimental.block_new_chunks('data_node_4', 'disttabl
 SELECT * FROM timescaledb_experimental.allow_new_chunks('data_node_4', 'disttable_3');
 \set ON_ERROR_STOP 1
 
--- detach table(s) where user has permissions, otherwise show NOTICE
-SELECT * FROM detach_data_node('data_node_4');
+-- detach table(s) where user has permissions, otherwise show
+-- NOTICE. Drop hypertables on data nodes in the process. Drop a
+-- hypertable on a data node manually to ensure the command can handle
+-- non-existing hypertables.
+CREATE TABLE disttable_5(time timestamptz, device int, temp float);
+SELECT * FROM create_distributed_hypertable('disttable_5', 'time', replication_factor => 1, data_nodes => '{"data_node_4", "data_node_5" }');
+SELECT * FROM test.remote_exec('{ data_node_4 }', $$ SELECT hypertable_name, owner FROM timescaledb_information.hypertables; $$);
+CALL distributed_exec($$ DROP TABLE disttable_4 $$, '{ data_node_4 }');
+SELECT * FROM detach_data_node('data_node_4', drop_hypertable => true);
```

Comment:
Do you have a test case for when the user does not have permissions to drop the table (for no permissions on the access node as well as permissions on the access node but not on the data node)? It is not a normal situation, but verifying that it works is always prudent.

Reply:
Maybe `drop_remote_data` is a better parameter name? But yes, I guess it is possible to also create an artificial test where the user doesn't have permissions on the DN to drop the hypertable.

Reply:
I'm referring to lack of privileges to drop the table on the data node, not the access node. Sorry if that was unclear.

I am thinking about two scenarios in particular:

1. You can modify the access node but not make the data node reflect the same state, which could cause a strange state if possible.
2. You can modify the data node but not the access node, which if done in the wrong order would mean that the access node could refer to something that does not exist any more.

Reply:
@mkindahl Added a new permission-based test although I am not sure this is what you wanted.

Reply:
> Added a new permission-based test although I am not sure this is what you wanted.

Thanks!

> Are you worrying that the permission system will not work as it is supposed to or that the error is not returned from the data node? (All of this is tested elsewhere separately).

Distributed systems can accidentally or explicitly be non-transactional, which means that errors in one part does not necessarily propagate to other parts, so I am want to ensure that an error on the data node (or the access node) does not leave the system in a strange state neither now nor in the future.

> I mean, there's an infinite number of ways to artificially mess up the system and thus fail this command. 

Yes, but the most common mistakes we make is having wrong permissions on the data node and not taking into consideration that objects might not exist on the data node, so worthwhile to test these cases.

Reply:
FWIW, we do test error propagation elsewhere. IMO, if we have to retest error propagation every time we do a remote op in a test (by inducing artificial errors), then tests will inflate and be unwieldy to maintain and understand. It is better to explicitly test error propagation in a dedicated test in that case.